rest-service/manager_rest/storage/storage_manager.py

```diff
@@ -629,3 +673,95 @@ def __iter__(self):
 
     def __getitem__(self, item):
         return self.items[item]
+
+
+class SQLAlchemyTracingHelper(object):
+    @staticmethod
+    def enable_tracing_for_all_engines():
+        listen(Engine, 'before_cursor_execute',
+               SQLAlchemyTracingHelper._engine_before_cursor_handler)
+        listen(Engine, 'after_cursor_execute',
+               SQLAlchemyTracingHelper._engine_after_cursor_handler)
+        listen(Engine, 'handle_error',
+               SQLAlchemyTracingHelper._engine_error_handler)
+
+    @staticmethod
+    def _clear_traced(obj):
+        '''
+        Clear an object's decorated tracing fields,
+        to prevent unintended further tracing.
+        '''
+        if hasattr(obj, '_parent_span'):
+            del obj._parent_span
+        if hasattr(obj, '_traced'):
+            del obj._traced
+
+    @staticmethod
+    def _normalize_stmt(statement):
+        return statement.strip().replace('\n', '').replace('\t', '')
+
+    @staticmethod
+    def _engine_before_cursor_handler(conn, cursor,
+                                      statement, parameters,
+                                      context, executemany):
+        current_span = get_current_span()
+        if not current_span:
+            return
+
+        stmt_obj = None
+        if context.compiled is not None:
+            stmt_obj = context.compiled.statement
+
+        # Don't trace PRAGMA statements coming from SQLite
+        if stmt_obj is None and statement.startswith('PRAGMA'):
+            return
+
+        # Start a new span for this query.
+        name = SQLAlchemyTracingHelper._get_operation_name(stmt_obj)
+        span = current_app.tracer.start_span(
+            operation_name=name, child_of=current_span)
+        span.set_tag('component', 'sqlalchemy')
+        span.set_tag('db.type', 'sql')
+        span.set_tag('db.statement',
+                     SQLAlchemyTracingHelper._normalize_stmt(statement))
+        span.set_tag('sqlalchemy.dialect', context.dialect.name)
+
+        context._span = span
+
+    @staticmethod
+    def _engine_after_cursor_handler(conn, cursor,
+                                     statement, parameters,
+                                     context, executemany):
+        span = getattr(context, '_span', None)
+        if span is None:
+            return
+
+        span.finish()
+
+        if context.compiled is not None:
```

Comment:
Add why this is needed

Reply:
Forgot to remove these leftovers. Removed.